File: my-vue-app/src/api/modules/game.ts
import request from '@/utils/request'
import type { GameState } from '@/types/game'

export const startGameApi = (topic: string) => {
    return request.post<any, GameState>('/game/start', { topic })
}

export const nextStepApi = () => {
    return request.post<any, GameState>('/game/next')
}

export const getGameStatusApi = () => {
    return request.get<any, GameState>('/game/status')
}

File: backend/app/api/v1/endpoints/game.py
from fastapi import APIRouter, Body
from app.services.game_service import GameService
from app.schemas.game import GameState

router = APIRouter()

@router.post("/start", response_model=GameState)
def start_game(topic: str = Body(..., embed=True)):
    """
    初始化并开始新游戏
    """
    return GameService.start_game(topic)

@router.post("/next", response_model=GameState)
def next_step():
    """
    推进游戏一步
    """
    return GameService.next_step()

@router.get("/status", response_model=GameState)
def get_game_status():
    """
    获取当前游戏状态
    """
    return GameService.get_game_state()
